headers/private/kernel/arch/x86/64/descriptors.h

```diff
@@ -43,6 +43,11 @@ struct segment_descriptor {
 	uint32 base1 : 8;
 } _PACKED;
 
+
+#ifndef __cplusplus
+typedef struct segment_descriptor segment_descriptor;
```

Comment:
probably belongs in arch_thread_types (or whatever header this is natively declared in.)

Reply:
What do you mean by "header this is natively declared in"?
Do you want the typedefs (both this one and the one at cpu.h) to go into arch_cpu.h directly?

Reply:
Sorry but I don't quite get it. What do you mean by "this" ? Do you mean we should have the typedef both in C++ and C mode?

Maybe it's just better to rename every "segment_descriptor" by "struct segment_descriptor", which is how having both C++ and C apis is done in some parts of the codebase.

Reply:
Ah, I was confused, I see now that `segment_descriptor` is declared in this file.

I think you can enable this in C++ mode too, though; it should be harmless.